1-weekly-mission-week7-main/src/App.js

```diff
@@ -0,0 +1,33 @@
+import { Layout } from "./feature/Layout";
+import { SharedPage } from "./page-layout/SharedPage";
+import "./global.css";
+import { FolderInfo } from "./ui/FolderInfo";
+import { SearchBar } from "./ui/SearchBar";
+import { CardList } from "./ui/CardList";
+import { useGetFolder } from "data-access/useGetFolder";
+import { ReadOnlyCard } from "ui/ReadOnlyCard";
+
+function App() {
+  const { data } = useGetFolder();
+  const { profileImage, ownerName, folderName, links } = data || {};
+
+  return (
+    <Layout>
+      <SharedPage
+        folderInfo={
+          <FolderInfo profileImage={profileImage} ownerName={ownerName} folderName={folderName} />
+        }
+        searchBar={<SearchBar />}
+        cardList={
+          <CardList>
+            {links?.map((link) => (
+              <ReadOnlyCard key={link?.id} {...link} />
+            ))}
+          </CardList>
+        }
+      />
```

Comment:
💊
코드에서 `SharedPage` 컴포넌트에 `folderInfo`, `searchBar`, `cardList`를 `props`로 전달하는 방식으로 렌더 패턴을 사용하고 있는데요. 

이는 컴포넌트의 유연성을 높이고 재사용성을 개선할 수 있는 방법이지만, 과도하게 사용될 경우 코드의 가독성과 유지보수성에 영향을 줄 수 있습니다.
 
**특히, 각각의 컴포넌트가 SharedPage 내에서만 사용되고, 복잡한 로직이나 상태를 관리하지 않는다면, 이러한 방식은 오버 엔지니어링으로 볼 수 있습니다.**

```suggestion
    <SharedPage>
        <FolderInfo profileImage={profileImage} ownerName={ownerName} folderName={folderName} />
        <SearchBar />
        <CardList links={links} />
      </SharedPage>
```

이런식으로 컴포넌트 내에서 직접 필요한 컴포넌트를 렌더링하고, 필요한 데이터를 props로 전달하는 방식으로 구조를 단순화 할 수 있을 것 같은데 어떠신가요?? 👀

Reply:
만약 이러한 render pattern을 사용하고 싶으시다면 이 상황에선 Compound Component패턴을 사용하는 펀이 더 좋을 수도 있을 것 같아요 (물론 여기선 오버엔지니어링이라 비추천합니다)

참고 : https://www.youtube.com/watch?v=vPRdY87_SH0